### PR TITLE
makedown docs for some API

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request

checkin makedown docs, includes:

(1) 11 nn layers:
Index,
L1HingeEmbeddingCriterion,
Clamp,
LSTMPeephole,
SoftPlus,
Bottle,
SoftmaxWithCriterion,
AbsCriterion,
CSubTable,
CMulTable,
LogSoftMax,

(2) 1 optim method
ValidationMethod

Not sure:
whether it is need to check model.backward

### CODE REVIEW

File: docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+## AbsCriterion ##
+
+**Scala:**
+```scala
+val criterion = AbsCriterion[Float]()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md]: AbsCriterion[T]

File: docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+```
+**Python:**
+```python
+criterion = AbsCriterion()
+```
+
+measures the mean absolute value of the element-wise difference between input
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md]: Measures

Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md]: between input and output?

File: docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+val criterion = AbsCriterion[Float]()
+
+val input = Tensor[Float](3).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md]: don't random here. It's not easy to read. Tensor[Float](T(1.0f, 2.0f, 3.0f))

File: docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+val target = Tensor[Float](3).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md]: don't random here. It's not easy to read. Tensor[Float](T(1.0f, 2.0f, 3.0f))

File: docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+val output = criterion.forward(input, target)
+```
+output is
+```
+output: Float = 0.33056465
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md]: This is meanless. Users still can't refer how do you get the output. They don't know the inputs

File: docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+**Python example:**
+```python
+criterion = AbsCriterion()
+input = np.array([0.9682213801388531,
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/AbsCriterion.md]: use simpler number

File: docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+## L1HingeEmbeddingCriterion ##
+
+**Scala:**
+```scala
+val model = L1HingeEmbeddingCriterion[Float](margin)
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md]: [T]

File: docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+a table of two Tensors, and a label y (1 or -1):
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T
+
+val model = L1HingeEmbeddingCriterion[Float](0.6)
+val input1 = Tensor[Float](2).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md]: don't random

File: docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+```
+output is
+```
+output: Float = 0.84714425
+```
+
+**Python example:**
+```python
+model = L1HingeEmbeddingCriterion(0.6)
+input1 = np.random.randn(2)
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md]: don't random

File: docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+**Scala:**
+```scala
+val model = L1HingeEmbeddingCriterion[Float](margin)
+```
+**Python:**
+```python
+model = L1HingeEmbeddingCriterion(margin)
+```
+
+Creates a criterion that measures the loss given an input ``` x = {x1, x2} ```,
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/L1HingeEmbeddingCriterion.md]: I don't understand the description

File: docs/docs/APIdocs/layers/nn/Loss Functions/SoftmaxWithCriterion.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+```scala
+val model = SoftmaxWithCriterion[T](ignoreLabel, normalizeMode)
+```
+**Python:**
+```python
+model = SoftmaxWithCriterion(ignoreLabel, normalizeMode)
+```
+
+Computes the multinomial logistic loss for a one-of-many classification task,
+passing real-valued predictions through a softmax to get a probability distribution over classes.
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/SoftmaxWithCriterion.md]: what's the shape of the inputs? what's the real-valued predictions?

File: docs/docs/APIdocs/layers/nn/Loss Functions/SoftmaxWithCriterion.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+as its gradient computation is more numerically stable.
+- param ignoreLabel   (optional) Specify a label value that should be ignored when computing the loss.
+- param normalizeMode How to normalize the output loss.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+
+val input = Tensor[Float](1, 5, 2, 3).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/SoftmaxWithCriterion.md]: don't random

File: docs/docs/APIdocs/layers/nn/Loss Functions/SoftmaxWithCriterion.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+val target = Tensor(Storage(Array(2.0f, 4.0f, 2.0f, 4.0f, 1.0f, 2.0f))).resize(1, 1, 2, 3)
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Loss Functions/SoftmaxWithCriterion.md]: Tensor[Float](T(T(
  T(2.0f, 4.0f, 2.0f),
  T(4.0f, 1.0f, 2.0f)
)))

File: docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+Bottle allows varying dimensionality input to be forwarded through any module that accepts input of nInputDim dimensions, and generates output of nOutputDim dimensions.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+
+val model = Bottle[Float](Linear[Float](10, 2), 2, 2)
+model.add(Linear(10, 2))
+val input = Tensor[Float](4, 5, 10).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md]: Don't random

File: docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+0.8234116	-0.6176827	
+
+(4,.,.) =
+0.8869035	-0.51233184	
+0.65199244	-0.48857856	
+0.7880871	-0.7456757	
+0.8663832	-0.22757408	
+0.9411352	-0.8008182	
+
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 4x5x2]
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md]: output is too long. Can you make it shorter with other input?

File: docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+**Scala:**
+```scala
+val model = Bottle[T](module, nInputDim, nOutputDim)
+```
+**Python:**
+```python
+model = Bottle(module, nInputDim, nOutputDim)
+```
+
+Bottle allows varying dimensionality input to be forwarded through any module that accepts input of nInputDim dimensions, and generates output of nOutputDim dimensions.
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md]: varying dimensionality input. Can you explain more?

File: docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+
+val model = Bottle[Float](Linear[Float](10, 2), 2, 2)
+model.add(Linear(10, 2))
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Bottle.md]: there're two Linear?

File: docs/docs/APIdocs/layers/nn/Simple Layers/CMulTable.md [hunk 18]
```diff
@@ -0,0 +1,49 @@
+## CMulTable ##
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/CMulTable.md]: Put it to Merge Layer

File: docs/docs/APIdocs/layers/nn/Simple Layers/CMulTable.md [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+**Scala:**
+```scala
+val model = CMulTable[T]()
+```
+**Python:**
+```python
+model = CMulTable()
+```
+
+Takes a table of Tensors and outputs the multiplication of all of them.
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/CMulTable.md]: What is a table? change it to sequence

File: docs/docs/APIdocs/layers/nn/Simple Layers/CMulTable.md [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T
+
+val model = CMulTable[Float]()
+
+val input1 = Tensor[Float](5).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/CMulTable.md]: don't random

File: docs/docs/APIdocs/layers/nn/Simple Layers/CSubTable.md [hunk 21]
```diff
@@ -0,0 +1,49 @@
+## CSubTable ##
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/CSubTable.md]: put it to merge layer

File: docs/docs/APIdocs/layers/nn/Simple Layers/CSubTable.md [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+**Scala:**
+```scala
+val model = CSubTable[T]()
+```
+**Python:**
+```python
+model = CSubTable()
+```
+
+Takes a table with two Tensor and returns the component-wise subtraction between them.
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/CSubTable.md]: Table -> Sequence

File: docs/docs/APIdocs/layers/nn/Simple Layers/CSubTable.md [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T
+
+val model = CSubTable[Float]()
+
+val input1 = Tensor[Float](5).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/CSubTable.md]: don't random

File: docs/docs/APIdocs/layers/nn/Simple Layers/Clamp.md [hunk 24]
```diff
@@ -0,0 +1,52 @@
+## Clamp ##
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Clamp.md]: Don't you read the document of the layer...? Obvious it's a non-linear activation?

File: docs/docs/APIdocs/layers/nn/Simple Layers/Clamp.md [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+**Scala:**
+```scala
+val model = Clamp[T](min, max)
+```
+**Python:**
+```python
+model = Clamp(min, max)
+```
+
+A kind of hard tanh activition function with integer min and max
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Clamp.md]: Are you sure the document is right? https://github.com/torch/nn/blob/master/doc/simple.md#nn.Clamp

File: docs/docs/APIdocs/layers/nn/Simple Layers/Clamp.md [hunk 26]
```diff
@@ -0,0 +1,52 @@
...
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+
+val model = Clamp[Float](-10, 10)
+val input = Tensor[Float](2, 2, 2).rand()
+val output = model.forward(input)
+```
+output is
+```
+output: com.intel.analytics.bigdl.tensor.Tensor[Float] =
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Clamp.md]: where's your input

File: docs/docs/APIdocs/layers/nn/Simple Layers/Index.md [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+
+Applies the Tensor index operation along the given dimension.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T
+
+val input1 = Tensor[Float](3).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/Index.md]: don't rand.

File: docs/docs/APIdocs/layers/nn/Simple Layers/LSTMPeephole.md [hunk 28]
```diff
@@ -0,0 +1,90 @@
+## LSTMPeephole ##
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/LSTMPeephole.md]: Simple Layer? Are you sure?

File: docs/docs/APIdocs/layers/nn/Simple Layers/LSTMPeephole.md [hunk 29]
```diff
@@ -0,0 +1,90 @@
...
+
+**Scala:**
+```scala
+val model = LSTMPeephole[Float](inputSize, hiddenSize)
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/LSTMPeephole.md]: Float -> T. Why even in your own PR, this is not consisitent

File: docs/docs/APIdocs/layers/nn/Simple Layers/LSTMPeephole.md [hunk 30]
```diff
@@ -0,0 +1,90 @@
...
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.RandomGenerator._
+
+val hiddenSize = 4
+val inputSize = 6
+val outputSize = 5
+val seqLength = 5
+val batchSize = 1
+               
+val input = Tensor[Float](Array(batchSize, seqLength, inputSize))
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/LSTMPeephole.md]: don't random

File: docs/docs/APIdocs/layers/nn/Simple Layers/LogSoftMax.md [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+**Scala:**
+```scala
+val model = LogSoftMax[T]()
+```
+**Python:**
+```python
+model = LogSoftMax()
+```
+
+The [[LogSoftMax]] module applies a LogSoftMax transformation to the input data
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/LogSoftMax.md]: why [[]]?

File: docs/docs/APIdocs/layers/nn/Simple Layers/LogSoftMax.md [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+The input given in `forward(input)` must be either
+a vector (1D tensor) or matrix (2D tensor).
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+
+val model = LogSoftMax[Float]()
+val input = Tensor[Float](4, 10).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/LogSoftMax.md]: don't random

File: docs/docs/APIdocs/layers/nn/Simple Layers/LogSoftMax.md [hunk 33]
```diff
@@ -0,0 +1,56 @@
...
+val output = model.forward(input)
+```
+output is
+```
+output: com.intel.analytics.bigdl.tensor.Tensor[Float] =
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/LogSoftMax.md]: output is too large

File: docs/docs/APIdocs/layers/nn/Simple Layers/SoftPlus.md [hunk 34]
```diff
@@ -0,0 +1,58 @@
+## SoftPlus ##
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/SoftPlus.md]: simple layer???

File: docs/docs/APIdocs/layers/nn/Simple Layers/SoftPlus.md [hunk 35]
```diff
@@ -0,0 +1,58 @@
...
+```
+- param beta Controls sharpness of transfer function
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+
+val model = SoftPlus[Float]()
+val input = Tensor[Float](2, 3, 4).rand()
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/SoftPlus.md]: don't random

File: docs/docs/APIdocs/layers/nn/Simple Layers/SoftPlus.md [hunk 36]
```diff
@@ -0,0 +1,58 @@
...
+val output = model.forward(input)
+```
+output is
+```
+output: com.intel.analytics.bigdl.tensor.Tensor[Float] = 
+(1,.,.) =
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/nn/Simple Layers/SoftPlus.md]: shorten the output

File: docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md [hunk 37]
```diff
@@ -0,0 +1,29 @@
+## ValidationMethod ##
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md]: move it outside the Metrics

File: docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md [hunk 38]
```diff
@@ -0,0 +1,29 @@
+## ValidationMethod ##
+
+ValidationMethod is defined to evaluate the model.
+This trait can be extended by user-defined method. Such as Top1Accuracy, Top5Accuracy.
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md]: what is trait?

File: docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md [hunk 39]
```diff
@@ -0,0 +1,29 @@
+## ValidationMethod ##
+
+ValidationMethod is defined to evaluate the model.
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md]: Explain more when it should be used

File: docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md [hunk 40]
```diff
@@ -0,0 +1,29 @@
+## ValidationMethod ##
+
+ValidationMethod is defined to evaluate the model.
+This trait can be extended by user-defined method. Such as Top1Accuracy, Top5Accuracy.
+
+Example code:
+
+```
+val output = Tensor(Storage(Array[Float](
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md]: Tensor(T(...))

File: docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md [hunk 41]
```diff
@@ -0,0 +1,29 @@
+## ValidationMethod ##
+
+ValidationMethod is defined to evaluate the model.
+This trait can be extended by user-defined method. Such as Top1Accuracy, Top5Accuracy.
+
+Example code:
+
+```
+val output = Tensor(Storage(Array[Float](
+      0, 0, 0, 1,
```
Comment (yiheng) [inline docs/docs/APIdocs/layers/optim/Metrics/ValidationMethod.md]: This is just top1, what about top5

**Comment by yiheng** [comment]
API makedown???

**Comment by yiheng** [comment]
Where's the backward example???

**Comment by yiheng** [comment]
Please add more description to your PR

**Comment by zhangxiaoli73** [comment]
this makedown includes:
(1) 12 nn layers:
Index,
L1HingeEmbeddingCriterion,
Clamp,
LSTMPeephole,
SoftPlus,
Bottle,
SoftmaxWithCriterion,
AbsCriterion,
CSubTable,
CMulTable,
LogSoftMax,
(2) 1 optim method
ValidationMethod

**Comment by zhangxiaoli73** [comment]
jenkins pass

**Comment by i8run** [comment]
LGTM

**Pull request merged**